### DISCUSSION TITLE
502 bad gateway

### DISCUSSION THREAD

**Discussion by hle5128** [SmoothWAN/SmoothWAN#32] [General & Help]
I sometimes get a 502 lousy gateway in Luci and need to reboot to bring back the admin.
Do you know the reasons?
<img width="1404" alt="Screen Shot 2022-05-03 at 11 53 27 AM" src="https://user-images.githubusercontent.com/8794968/166489556-73e79682-e255-4d51-9fc9-7dd348a27ed2.png">

**Comment by hle5128** [comment]
can not get back to admin, the internet is still up, also can not ssh as well when this happens (permission denied on root)

**Reply by TalalMash** [reply]
Hello, this sounds like sdcard issue or corruption, try reflashing or changing sdcard model/brand.

**Reply by hle5128** [reply]
I'm using SSD card booting, reboot does let me back in though,
i dont have problem with this in the past with ras pi 4

**Reply by TalalMash** [reply]
> I'm using SSD card booting, reboot does let me back in though, i dont have problem with this in the past with ras pi 4

I see.. There is limited root disk detection in OpenWRT, and the additional filesystem check script targets sdcards. Maybe it's in read-only mode due to some error? Please post the last 30 lines or so of `dmesg` output here, make sure you open an SSH session before it errors again.

You have to modify the filesystem check-repair to work with SSD by editing in terminal the file /lib/preinit/80_mount_root and change `/dev/mmcblk0p2` to `/dev/sda2` at line 21.

**Reply by hle5128** [reply]
yup, i have done that, will get the logs for you next time. thanks

**Reply by TalalMash** [reply]
@hle5128 Thanks for the tip! I've moved this to issues.
